From 84fcf22f5689dace9d3d258b5beb9882629615d8 Mon Sep 17 00:00:00 2001
From: Jeremy Drake <github@jdrake.com>
Date: Mon, 14 Jun 2021 10:24:03 -0700
Subject: [PATCH 2/3] cast time_t to size_t for printf

the format string was using %zu, which expects its argument to be a
size_t, but time_t may be a different type, triggering for example
-Wformat or -Wformat-signedness.
---
 src/pacrepairfile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pacrepairfile.c b/src/pacrepairfile.c
index 5ed3454..4adedad 100644
--- a/src/pacrepairfile.c
+++ b/src/pacrepairfile.c
@@ -195,7 +195,7 @@ int fix_mtime(const char *path, struct archive_entry *entry) {
         path, strerror(errno));
     return 1;
   } else if (verbose) {
-    printf("%s: set modification time to %zu\n", path, t);
+    printf("%s: set modification time to %zu\n", path, (size_t)t);
   }
 
   return 0;
-- 
2.32.0.windows.1

